src/main/java/org/symphonyoss/symphony/messageml/elements/Checkbox.java

```diff
@@ -87,23 +89,29 @@ protected void buildAttribute(org.w3c.dom.Node item) throws InvalidInputExceptio
 
   @Override
   public void asPresentationML(XmlPrintStream out) {
-    out.openElement(PRESENTATIONML_DIV_TAG, PRESENTATIONML_CLASS_ATTR, PRESENTATIONML_DIV_CLASS);
-
     Map<String, String> presentationAttrs = buildCheckboxInputAttributes();
-    out.printElement(INPUT_TAG, presentationAttrs);
-
-    out.openElement(PRESENTATIONML_LABEL_TAG);
-    for (Element child : getChildren()) {
-      child.asPresentationML(out);
+    
+    if (getChildren().isEmpty()) {
+      out.printElement(INPUT_TAG, presentationAttrs);
     }
-    out.closeElement(); // Closing label
+    else {
+      out.openElement(PRESENTATIONML_DIV_TAG, PRESENTATIONML_CLASS_ATTR, PRESENTATIONML_DIV_CLASS);
+      
+      out.printElement(INPUT_TAG, presentationAttrs);
+
+      out.openElement(PRESENTATIONML_LABEL_TAG);
+      for (Element child : getChildren()) {
+        child.asPresentationML(out);
+      }
+      out.closeElement(); // Closing label
 
-    out.closeElement(); // Closing div
+      out.closeElement(); // Closing div
+    }
   }
 
   @Override
   public Node asMarkdown() {
-    return new CheckboxNode();
+    return new CheckboxNode(getAttribute(NAME_ATTR));
```

Comment:
After reading the remaining code I understood the way you were handling markdown, however, that's not the correct way. You should instead be creating FormElementNode directly in here and passing the "name" attribute as the "text" of the node itself, instead of concatenating the constant with the name of the field later on.